src/main/java/com/github/jikoo/anvilunlocker/AnvilUnlocker.java

```diff
@@ -22,9 +22,10 @@ public void onEnable() {
 
 	@EventHandler
 	public void onInventoryOpen(InventoryOpenEvent event) {
+		int maximumCost = getConfig().getInt("maximumcost");
 		if (event.getInventory() instanceof AnvilInventory && event.getPlayer() instanceof Player
 				&& event.getPlayer().getGameMode() != GameMode.CREATIVE) {
-			((AnvilInventory) event.getInventory()).setMaximumRepairCost(Short.MAX_VALUE);
+			((AnvilInventory) event.getInventory()).setMaximumRepairCost(maximumCost);
```

Comment:
Does this actually work? I imagine we need to toggle faked gamemode post-PrepareAnvilEvent state so that we're displaying "Too Expensive!" when the result is too expensive. The result should be denied, but does the client understand that? I imagine it will create some desync issues.

Reply:
After experimenting I found that the lack of "Too Expensive!" was very confusing for an otherwise legal combination, so the post-PrepareAnvilEvent toggle is in fact necessary.